### ISSUE TITLE
TElocal for transcript information?

### ISSUE THREAD

**Issue by sanhe374** [mhammell-laboratory/TElocal#35]
I have been using TEcount for estimating TEs in total RNA-seq data which has worked great. However, I want to combine this with methylation data and there we have methylation differences at the transcript level. Would running TElocal give me the transcript information for TEs and genes?

**Comment by sanhe374** [comment]
Thank you, where can I download the compressed tarball for TElocal?

**Comment by olivertam** [comment]
Hi,

The latest release is [here](https://github.com/mhammell-laboratory/TElocal/releases/tag/1.1.2)
You can also clone the repository with git.

Thanks

**Comment by sanhe374** [comment]
Thank you

**Comment by sanhe374** [comment]
I have an additional follow up on this thread. I have used hg38_rmsk_TE.gtf.locInd for running TElocal on my RNA-seq data. In the corresponding DNA methylation data, I have used the hg38_rmsk_TE.gtf. Both files were downloaded from your homepage.  Would this be the correct way of doing it to ensure that I am including the same TEs as used for TElocal?

**Comment by olivertam** [comment]
Hi,

The `hg38_rmsk_TE.gtf` is what is used to generate `hg38_rmsk_TE.gtf.locInd`, so the `transcript_id` should be the same. If you need confirmation, the genomic locations of the TE in `hg38_rmsk_TE.gtf.locInd` is located [here](https://labshare.cshl.edu/shares/mhammelllab/www-data/TElocal/annotation_tables/hg38_rmsk_TE.gtf.locInd.locations.gz).

Thanks.

**Comment by sanhe374** [comment]
I have tried to check the locations of hg38_rmsk_TE.gtf.locInd as compared to hg38_rmsk_TE.gtf. In the hg38_rmsk_TE.gtf.locInd there are 4693511 rows whereas in the hg38_rmsk_TE.gtf there are 4832405. Is there a reason why they are not the same numbers?

**Comment by olivertam** [comment]
Hi,

Thank you for your question. This is because the GTF contains patch chromosomes (they have the `_fix` in their name), whereas the `locInd` file do not. We chose this approach because we do not feel that the patch chromosomes tend to confuse most aligners.

Thanks.

**Comment by sanhe374** [comment]
Thank you. It appears that for some of TEs there is a a mismatch between the hg38_rmsk_TE.gtf. and hg38_rmsk_TE.gtf.locInd .location for example MIR3_dup86431 that appears on chr6_GL000254v2_alt	UCSC_rmsk	exon	3827134	3827293	231 in the  hg38_rmsk_TE.gtf. but on MIR3_dup86431	chr15_KI270849v1_alt:239916-239974:-
 in locations file. Is there are reason for this?

**Comment by olivertam** [comment]
Hi,

Ah yes, I suspected this. The hg38 TE GTF was probably updated recently. It would appear that the locInd was generated from an older hg38 build (i.e. less patches), and so the numbering might be different.
I can re-generate the locInd and annotation table to make them match if that's what you need. It would take a bit of time though.

Thanks.

**Comment by sanhe374** [comment]
Is it easier if I can get the older version of the hg38 TE GTF build? That way I do not need to redo TElocal.

Från: Oliver Tam ***@***.***>
Skickat: den 7 augusti 2024 14:13
Till: mhammell-laboratory/TElocal ***@***.***>
Kopia: Sandra Hellberg ***@***.***>; State change ***@***.***>
Ämne: Re: [mhammell-laboratory/TElocal] TElocal for transcript information? (Issue #35)


Hi,

Ah yes, I suspected this. The hg38 TE GTF was probably updated recently. It would appear that the locInd was generated from an older hg38 build (i.e. less patches), and so the numbering might be different.
I can re-generate the locInd and annotation table to make them match if that's what you need. It would take a bit of time though.

Thanks.

-

**Comment by sanhe374** [comment]
The most important thing is that I know that the hg38_rmsk_TE, hg38_rmsk_TE.loclnd and hg38_rmsk_TE.loclnd.locations are derived from the same initial build.

**Comment by olivertam** [comment]
Hi,

Let me take a look.
Thanks.

Cheers,
Oliver

**Comment by olivertam** [comment]
Hi,

Try this [file](https://labshare.cshl.edu/shares/mhammelllab/www-data/TEtranscripts/TE_GTF/archive/hg38_rmsk_TE.gtf.bak.gz).

Thanks.

**Comment by sanhe374** [comment]
It seems like the number of rows are the same, will take a closer look. How can I be sure that the loclnd used in TElocal is derived from the same?

**Comment by olivertam** [comment]
Hi,

I checked that the `MIR3_dup86431` is located on `chr15_KI270849v1_alt:239916-239974:-` in the GTF.
I have also converted the old GTF into the locInd locations file, and it has no difference (using `diff`) with the locInd locations file that is downloadable from the website.
Hope this is sufficient. Let me know if you need additional checks.

Thanks.

**Comment by sanhe374** [comment]
Thank you for checking! And the locInd locations file is derived from the hg38_rmsk_TE.loclnd that is derived from the old GTF?

**Comment by olivertam** [comment]
Hi,

They should be.
As I mentioned above, I'd be happy to confirm this by matching up the output with the GTF if you want me to send me a `TElocal` output file.

Thanks.

**Comment by sanhe374** [comment]
How can I send you the file?

**Comment by olivertam** [comment]
An addendum: I just checked an output file that I ran using the hg38 `locInd` file available online, and it matches up (in terms of number of entries, as well as the md5sum).

One quick way to check is that if you do the following:
```
$ grep ":" [TElocal output] | wc -l
4693511
```
It should match the number of entries from the GTF and locations file.

Otherwise, feel free to compress the file and share it via Dropbox (or other file sharing program).

Thanks.
